test/gpu/jit.cpp

```diff
@@ -356,4 +356,71 @@ TEST_CASE(compile_math)
     });
 }
 
+// NOLINTNEXTLINE
+const std::string assert_template = R"__migraphx__(
+#include <migraphx/kernels/math.hpp>
+#include <migraphx/kernels/types.hpp>
+using namespace migraphx;
+extern "C" {
+__global__ void kernel(void*) 
+{
+    static_assert(numeric_max<${type}>() == ${max}, "");
+    static_assert(numeric_lowest<${type}>() == ${min}, "");
+}
+}
+
+int main() {}
+
+)__migraphx__";
+
+TEST_CASE(assert_type_min_max)
+{
+    std::vector<std::string> data_types;
+    migraphx::gpu::hip_compile_options options;
+    for(auto&& t : migraphx::shape::types())
+    {
+        if(contains({migraphx::shape::bool_type, migraphx::shape::tuple_type}, t))
+            continue;
+        auto name = migraphx::shape::cpp_type(t);
+        if(t == migraphx::shape::half_type)
+            name.insert(0, "migraphx::");
+
+        std::string min = "";
+        std::string max = "";
+        // Note 9223372036854775808 is a constant literal that is outside the range of long long
+        // type For the same reason, 18446744073709551616 needs postfix ULL to be parsed correctly
+        auto min_override_int64  = "(-9223372036854775807LL - 1)";
+        auto max_override_uint64 = "18446744073709551615ULL";
+
+        migraphx::shape::visit(t, [&](auto as) {
+            if(t == migraphx::shape::int64_type)
```

Comment:
I dont think you need `if` statements here. I believe adding `LL` and `ULL` unconditionally should work for all types.

Reply:
OKay, yes, it seems reasonable to always add `LL` for minimal and `ULL` for max. Then I'd still need to `if` based on `int64_type` to override with `min_override_int64` because in that scenario without override it will give a `-9223372036854775808` which the compiler is not happy about.

Reply:
> because in that scenario without override it will give a -9223372036854775808 which the compiler is not happy about.

What error do you get from the compiler?

Reply:
The error I get is: `error: integer literal is too large to be represented in a signed integer type, interpreting as unsigned [-Werror,-Wimplicitly-unsigned-literal]`. 

As far as I know `"(-9223372036854775807LL - 1)"` is a reasonable way to override it.

Besides, the need for `LL` and `ULL` really only exist in 64 bit integer types. I thought it is going to be more clear that we take them as special cases. 

Now, because of this compiler integer parsing limitations, does this make `as.min()` and `as.max()` problemetic for 64 bit integer as eventually, those are replaced as-is to string as `${max}` and `${min}`? It looks to be a caveat exposed by this new unit test we've written.

Reply:
Talked with @pfultz2 and inline the literal in ec8c252. Please feel free to provide any additional feedback.